### PR TITLE
Task/map/download kml

### PR DESCRIPTION
**Pull request by teamagam7**

### CODE REVIEW

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class DownloadKmlTask extends AsyncTask<Void, Void, Boolean> {
+
+    private final String KML_DOWNLOAD_URL = "https://www.google.com/maps/d/kml?mid=1BCG-BscjjSsxAARv6nirLViIddw&forcekml=1";
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: Extract the constants here to config

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            byte data[] = new byte[4096];
+            int count;
+            while ((count = input.read(data)) != -1) {
+                output.write(data, 0, count);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return true;
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: Think about a case that you have an exception?
What will happen? What will be the return value?

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    private Context mContext;
+
+    public DownloadKmlTask(Context context) {
+        mContext = context;
+    }
+
+    @Override
+    protected Boolean doInBackground(Void... voids) {
+        try {
+            URL url = new URL(KML_DOWNLOAD_URL);
```
Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: Extract this big method to some small logical methods

Comment (teamagam3) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: Like `saveFile()`, `getResponse()`

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java (hunk 1 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: done

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java (hunk 3 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: done

File: HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java (hunk 2 above)
Comment (teamagam7) [inline HeverTasteApp/app/src/main/java/com/tm7/hevertaste/DownloadKmlTask.java]: done

**Pull request merged**